gcloud-java-compute/src/main/java/com/google/gcloud/compute/ImageConfiguration.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.api.services.compute.model.Image;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Base class for Google Compute Engine image configuration. Use {@link DiskImageConfiguration} to
+ * create an image from an existing Google Compute Engine disk. Use {@link RawImageConfiguration} to
+ * create an image from a file stored in Google Cloud Storage.
+ */
+public abstract class ImageConfiguration implements Serializable {
```

Comment:
I think this would help but my fear is that it might be confusing with `sourceType`. Also how should we call the tokens? `RAW` and `DISK`? `RAW` is confusing as it's also the name of the only value for `sourceType`. Maybe something like `DISK` and `STORAGE`? Then should we also rename the `RawImageConfiguration` to `StorageImageConfiguration`?